### PR TITLE
For Issue#70 - Auth0 Style OAuth2 Flow

### PR DESCRIPTION
**Pull request by karthik-mediamath**

### CODE REVIEW

File: src/main/java/com/mediamath/terminalone/TerminalOne.java [hunk 1]
```diff
@@ -266,22 +241,27 @@ public String getAuthorizationUrl(String redirectUri, String apiKey) throws Clie
...
-			throw new ClientException(UNABLE_TO_GET_O_AUTH_TOKEN + oauthProblemException.getMessage());
-		}
+		logger.info("Authenticating via OAuth Auth0.");
+		Form form = tOneService.getOAuthFormData(username, password, clientId, clientSecret);
+		Response oauthResponse = connection.post(oauthTokenUrl, form, null);
+		parseLoginError(oauthResponse);
+		String response = oauthResponse.readEntity(String.class);
+		logger.info("OAuth Response as String:"+response);
+		OAuthResponse parsedOAuthResponse = parseOAuthResponse(response);
+		logger.info("Parsed OAuth Response:"+parsedOAuthResponse);
```
Comment (fsargent) [inline src/main/java/com/mediamath/terminalone/TerminalOne.java]: Do we want to log all these things? This auth stuff probably shouldn't get written to disk in use.

File: src/test/java/com/mediamath/terminalone/functional/PostFunctionalTestIT.java [hunk 2]
```diff
@@ -114,11 +115,13 @@ public void testJTerminalOneStringStringString() throws ClientException {
 		assertEquals(true, t1.isAuthenticated());
 	}
 
+	// TEST WILL FAIL UNTIL AUTHO WORKS IN PRODUCTION
 	@Test
 	public void testOauthTokenAuthentication() throws ClientException {
 		TerminalOne t1 = new TerminalOne();
-		t1.authenticate("rydgu76jyap9sjaj5vyqpbg7");
-
+		OAuthResponse oauthResponse = t1.getOAuthToken("karthik87.mit@gmail.com", "Pm_oauthtest", "kRmmCjUNSTPhYnOlYPcl7ltIkBSITAVM",
```
Comment (fsargent) [inline src/test/java/com/mediamath/terminalone/functional/PostFunctionalTestIT.java]: This is very reckless. You should scrub this immediately.

Comment (fsargent) [inline src/test/java/com/mediamath/terminalone/functional/PostFunctionalTestIT.java]: Just rotated this. But for the future, you should use something like dotenv to store these variables locally. https://github.com/hotmeatballsoup/dotenv-java

**Comment by fsargent** [comment]
Can you validate that you were able to get a token from auth0?

File: src/main/java/com/mediamath/terminalone/TerminalOne.java (hunk 1 above)
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/TerminalOne.java]: Sure, can remove this. Added this to know the status of the response.

File: src/test/java/com/mediamath/terminalone/functional/PostFunctionalTestIT.java (hunk 2 above)
Comment (karthik-mediamath) [inline src/test/java/com/mediamath/terminalone/functional/PostFunctionalTestIT.java]: Those are dummy credentials and wont work. For actual testing I will read these parameters from test.config file.

**Comment by karthik-mediamath** [comment]
Yes, I can validate that the response has the token. Do we need a test or manual validation is sufficient?

**Comment by karthik-mediamath** [comment]
Made the necessary changes. Can you review?

**Comment by fsargent** [comment]
The best thing to do would to have these variables source from .env or the enviornment somehow.

-		OAuthResponse oauthResponse = t1.getOAuthToken("username", "password", "clientId",

It's not helpful to have them simply be strings, waiting for errors.
Do they read from test.config right now in this implementation?

**Comment by karthik-mediamath** [comment]
Hi,

Yes they are read from test.config. 

> OAuthResponse oauthResponse = t1.getOAuthToken(user, password,oauthKey ,
> 				oauthSecret);
> 

Each parameter for testing is read from config as,

> user = testConfig.getProperty("t1.username");

and 

> test.config

 should be populated with right credentials for these tests to work.

**Comment by fsargent** [comment]
LGTM. Merge when ready. May want to consider updating the README.md too on how to use the OAuth2 flow, and when.

**Pull request merged**